zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -634,6 +643,13 @@ private void writeRecords() {
             // We write various type of records. The positions are always increasing and
             // incremented by 1 for one record (even in a batch), so we can count the amount
             // of written records via the lastWritten and now written position.
```

Comment:
:x: This comment is at the wrong position now, it explains the `amount` calculation we use for metrics.

:wrench: This made me realize, the way we calculate `amount` is not really correct because it doesn't account for records from other sources. Instead, we can just say `metrics.recordsWritten(pendingWrites.size())`

Reply:
Placed the comment in the right place again and changed `amount` and also changed the argument for `metrics.recordsWritten`.